test/integration/nighthawk_grpc_service.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+import os
+import socket
+import subprocess
+import sys
+import tempfile
+import threading
+import time
+from string import Template
+
+from common import IpVersion
+
+
+class NighthawkGrpcService(object):
+  """
+    Base class for running a server in a separate process.
+    """
+
+  def __init__(self, server_binary_path, server_ip, ip_version):
+    assert ip_version != IpVersion.UNKNOWN
+    self.ip_version = ip_version
```

Comment:
Not sure if enforced elsewhere, but since this is a new file and a new class - can we distinguish between public and private attributes? We can prefix all private attributes with a single underscore. E.g. self._ip_version.

Additionally we could document any public attributes in the docstring of the class by adding an "Attributes" section.

class NighthawkGrpcService(object):
  """ Base class for running a server in a separate process.

  Attributes:
    server_binary_path: A string, path to the server binary.
    ...
  """

Ideally we would mark as many of these private as possible.